### PR TITLE
Add initial support for Silicon Labs EFM32PG-STK3402A board

### PR DESCRIPTION
**Pull request by GilBenkoe**
MicroPython and the following samples were tested: 
* hello_world 
* basic/button 
* subsys/shell/shell

Most of the changes are taken from the EFM32WG-STK3800 board.

### CODE REVIEW

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
+.. _efm32pg_stk3402a:
+
+EFM32PG-STK3402A
+###############
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Title underlines need to be at least as long as the title.  Add another ``#``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+The EFM32PG-STK3402A includes an `J-Link`_ serial and debug adaptor built into the
+board. The adaptor provides:
+
+- A USB connection to the host computer, which exposes a Mass Storage and a
+  USB Serial Port.
+- A Serial Flash device, which implements the USB flash disk file storage.
+- A physical UART connection which is relayed over interface USB Serial port.
+
+Flashing an application to EFM32PG-STK3402A
+----------------------------------------
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Title underling too short, add a few more ``---`` 's

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  consumption of an application or prototype design.
+- Ultra low power 128x128 pixel Memory-LCD
+- 2 user buttons, 2 LEDs and a touch slider
+- Humidity, temperature, and inductive-capacitive metal sensor
+- On-board Segger J-Link USB debugger
+
+For more information about the EFM32PG SoC and EFM32PG-STK3402A board:
+
+- `EFM32PG Website`_
+- `EFM32PG Datasheet`_
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: These links are called EFM32PG12 Datasheet (and Reference Manual) at the bottom of the doc, so you're getting a reference error from the doc generation tools.

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
+.. _efm32pg_stk3402a:
+
+EFM32PG-STK3402A
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Would a more friendly title be, "EFM32 Pearl Gecko Starter Kit" ?

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+###############
+
+Overview
+********
+
+The EFM32 Pearl Gecko Starter Kit EFM32PG-STK3402A contains a MCU from the
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: ``contains an MCU``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+EFM32PG family built on ARM® Cortex®-M4F processor with excellent low
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: ``build on an ARM``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/efm32pg_stk3402a/efm32pg_stk3402a_defconfig``
+
+Other hardware features are currently not supported by the port.
+
+Connections and IOs
+===================
+
+The EFM32PG SoC has six gpio controllers (PORTA to PORTF), but only four are
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: ``GPIO``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+currently enabled (PORTA, PORTB, PORTD and PORTF) for the EFM32PG-STK3402A board.
+
+In the following table, the column Name contains Pin names. For example, PE2
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Make the column name bold by changing to ``**Name**``
Change ``Pin`` to ``pin`` here and on the next line.

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+   Before using the kit the first time, you should update the J-Link firmware
+   from `J-Link-Downloads`_
+
+Flashing
+========
+
+The EFM32PG-STK3402A includes an `J-Link`_ serial and debug adaptor built into the
+board. The adaptor provides:
+
+- A USB connection to the host computer, which exposes a Mass Storage and a
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: change to ``a mass storage device``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+  USB Serial Port.
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: change to ``serial port`` (lowercase)

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+- A Serial Flash device, which implements the USB flash disk file storage.
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: change to ``serial flash`` (lowercase)

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+- A physical UART connection which is relayed over interface USB Serial port.
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: change to ``serial`` (lowercase)

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+The sample application :ref:`hello_world` is used for this example.
+Build the Zephyr kernel and application:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: efm32pg_stk3402a
+   :goals: build
+
+Connect the EFM32PG-STK3402A to your host computer using the USB port and you
+should see a USB connection which exposes a Mass Storage (STK3402A) and a
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: change to ``a mass storage device``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+USB Serial Port. Copy the generated zephyr.bin in the STK3402A drive.
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: change to ``serial port``

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+Use for example a converter like FT232/CP2102 to connect to UART on the
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Change to:

    Use a USB-to-UART converter such as an FT232/CP2102 to connect to the UART on the
    expansion header.

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst [hunk 16]
```diff
@@ -0,0 +1,168 @@
...
+expansion header.
+
+Open a serial terminal (minicom, putty, etc.) with the following settings:
+
+- Speed: 115200
+- Data: 8 bits
+- Parity: None
+- Stop bits: 1
+
+Reset the board and you should be able to see on the corresponding Serial Port
```
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Change to:

    Reset the board and you'll see the following message on the corresponding serial port
    terminal session:

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for providing the documentation.  Here are some suggested edits for clarity and to fix some doc generation issues.

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 1 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 2 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 3 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 4 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done.

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 5 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 6 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 7 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 8 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 9 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 10 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 11 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 12 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 13 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 14 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 15 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

File: boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst (hunk 16 above)
Comment (GilBenkoe) [reply boards/arm/efm32pg_stk3402a/doc/efm32wg_stk3800.rst]: Done

**Review by dbkinder — APPROVED** [review]
Docs LGTM, thanks!

**Review by galak — CHANGES_REQUESTED** [review]
Can you add a README per: http://docs.zephyrproject.org/contribute/contribute_non-apache.html

**Comment by GilBenkoe** [comment]
Ok, I did.

**Comment by chrta** [comment]
Because commit 2eb347b174046037139c024e1ab2f63caf549315 is already merged, please keep in mind to delete `soc/arm/silabs_exx32/efm32pg/CMakeLists.txt` and `soc/arm/silabs_exx32/efr32pg/soc.c`

**Comment by ioannisg** [comment]
@galak , it seems to me this PR is abandoned; @GilBenkoe , please, advice otherwise :)

**Comment by chrta** [comment]
Once #9042 is merged, I will rebase and continue working on this.

**Comment by chrta** [comment]
I cannot push to the branch that is reviewed here. I updated everything and tested it on the hardware.
Branch https://github.com/chrta/zephyr/tree/add_silabs_efm32pg_stk3402a

Should i create a new pull request with my branch or is there another possibility?

**Comment by ioannisg** [comment]
> 
> 
> I cannot push to the branch that is reviewed here. I updated everything and tested it on the hardware.
> Branch https://github.com/chrta/zephyr/tree/add_silabs_efm32pg_stk3402a
> 
> Should i create a new pull request with my branch or is there another possibility?

As you prefer. If you close this PR is will be  hard to follow-up the re-review on the new branch.
I think you can still push to this PR branch (on  @GilBenkoe 's fork, though)

**Comment by chrta** [comment]
@ioannisg I would prefer to continue here, but i get the following error message:

```
$ LANG=C git push gilbenkoe add_silabs_efm32pg_stk3402a 
ERROR: Permission to GilBenkoe/zephyr.git denied to chrta.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by galak** [comment]
> @ioannisg I would prefer to continue here, but i get the following error message:

while in general I would agree, but since we can't push to GilBenkoe branch, lets just create a new PR and we can reference this one in that.

**Comment by galak** [comment]
Closing as this is replaced by #11065
